openedx/core/djangoapps/xmodule_django/models.py

```diff
@@ -181,6 +183,19 @@ class UsageKeyField(OpaqueKeyField):
     KEY_CLASS = UsageKey
 
 
+class UsageKeyWithRunField(UsageKeyField):
+    """
+    Subclass of UsageKeyField that automatically fills in
+    missing `run` values, for old Mongo courses.
+    """
+    def to_python(self, value):
+        value = super(UsageKeyWithRunField, self).to_python(value)
+        if value is not None:
+            assert isinstance(value, UsageKey)
```

Comment:
Does this `replace` call currently happen for all courses, Mongo or not? And if so, would it save any computation to check for the presence of a run before calling `replace`?